Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -241,12 +295,94 @@ def isImportsOnlyFile (lines : Array String) : Bool :=
+  if h : pos < s.endPos then
+    have := Nat.sub_lt_sub_left h (String.lt_next s pos)
+    let posₙ := s.next pos -- `pos` is valid by assumption. `pos` is not `endPos` by check above.
+    -- `posₙ` is valid, might be `endPos`.
+    let cₙ := s.get? posₙ |>.getD '\uFFFD' -- �
+    if cₙ == UnicodeVariant.emoji && !(emojis.contains c) then
+      -- bad: unwanted emoji-variant-selector
+      let errₙ := err.push (.unicodeVariant ⟨[c, cₙ]⟩ none pos)
+      findBadUnicodeAux s  cₙ errₙ posₙ
+    else if cₙ == UnicodeVariant.text && !(nonEmojis.contains c) then
+      -- bad: unwanted text-variant selector
+      let errₙ := err.push (.unicodeVariant ⟨[c, cₙ]⟩ none pos)
+      findBadUnicodeAux s  cₙ errₙ posₙ
+    else if cₙ != UnicodeVariant.emoji && emojis.contains c then
+      -- bad: missing emoji-variant selector
+      let errₙ := err.push (.unicodeVariant ⟨[c]⟩ UnicodeVariant.emoji pos)
+      findBadUnicodeAux s  cₙ errₙ posₙ
+    else if cₙ != UnicodeVariant.text && nonEmojis.contains c then
+      -- bad: missing text-variant selector
+      let errₙ := err.push (.unicodeVariant ⟨[c]⟩ UnicodeVariant.text pos)
+      findBadUnicodeAux s cₙ errₙ posₙ
+    else
+      -- okay
+      findBadUnicodeAux s cₙ err posₙ
+  -- emojis/non-emojis should not be the last character in the line
+  else if emojis.contains c || nonEmojis.contains c then
+    err.push (.unicodeVariant ⟨[c]⟩ none pos)
+  else
+    err
+termination_by s.endPos.1 - pos.1
+
+/-- Creates `StyleError`s for bad usage of (emoji/text)-variant-selectors. -/
+@[inline]
+def findBadUnicode (s : String) : Array StyleError :=
+  if s == "" then #[] else
+  let c := s.get 0
+  -- edge case: variant-selector as first char of the line
+  let initalErrors := if #[UnicodeVariant.emoji, UnicodeVariant.text].contains c then
+    #[(.unicodeVariant ⟨[c]⟩ none 0)] else #[]
+  findBadUnicodeAux s c initalErrors
+
+end UnicodeLinter
+
+/-- Lint a collection of input strings if one of them contains unwanted unicode. -/
+def unicodeLinter : TextbasedLinter := fun lines ↦ Id.run do
+  let mut changed : Array String := #[]
+  let mut errors : Array (StyleError × ℕ) := Array.mkEmpty 0
+  let mut lineNumber := 1
+  for line in lines do
+    let err := UnicodeLinter.findBadUnicode line
+
+    -- try to auto-fix the style error
+    let mut newLine := line
+    for e in err.reverse do -- reversing is a cheap fix to prevent shifting indices
+      match e with
+      | .unicodeVariant s sel pos =>
+        let head := newLine.extract 0 pos
+        let tail := newLine.extract (head ++ s).endPos line.endPos
+        newLine := match sel with
+        | some v =>
+          -- injecting desired variant-selector
+          head ++ ⟨[s.get 0, v]⟩ ++ tail
+        | none =>
+          -- removing used variant-selector
+          head ++ ⟨[s.get 0]⟩ ++ tail
+      | _ =>
+        unreachable!
+
+    changed := changed.push newLine
+    errors := errors.append (err.map (fun e => (e, lineNumber)))
```

Comment:
use \mapsto?

Reply:
Changed it to 

```
errors := errors.append <| err.map ((·, lineNumber))
```

in 7ff43f24afa1dffc4d1411bca2cc5245b33f623a and 6700b73478108dbed368be4464f04597aaf62320